dockstore-common/src/main/java/io/dockstore/common/yaml/DockstoreYamlHelper.java

```diff
@@ -177,7 +177,15 @@ private static DockstoreYaml12 readDockstoreYaml12(final String content) throws
             final Yaml yaml = new Yaml(constructor);
             return yaml.load(content);
         } catch (Exception e) {
-            final String msg = ERROR_READING_DOCKSTORE_YML + e.getMessage();
+            String msg = ERROR_READING_DOCKSTORE_YML;
+            String errorMsg = e.getMessage();
+            if (errorMsg.startsWith("Cannot create property=")) {
+                int startIndex = errorMsg.indexOf("=") + 1;
```

Comment:
Could use a test, these string parse indexes feel breakable if the underlying library changes

Reply:
Agreed. This class has a fair amount of unit tests, which are easier and less cumbersome to deal with than integration tests (not dissing integration tests by any means). But with unit tests, you can easily and quickly feed in a variety of test cases. I suggest adding tests in there.